### ISSUE TITLE
[Question] How can we add a disk to harvester cluster in order to use it as data storage?

### ISSUE THREAD

**Issue by KarlSK8** [harvester/harvester#2458]
I need to add a disk to harvester to use it as data storage because I don't want to store data on the default disk /var/lib/harvester/defaultdisk. I am using a Nimble SAN connection from HPE and want to connect it to harvester, to use it as the primary datastore of all the nodes in that cluster. Is that possible? If yes how is it done?
Please provide a step by step description in case you have face such issue and was able to work it out.
I have personally connected a SAN to harvester then created an ext4 partition that I was able to view in the add disk view when editing.
If I go to longhorn and add a disk using it's path, it says that the disk is not ready and not schedulable
![harv](https://user-images.githubusercontent.com/79142047/177569925-5cbe7864-2df5-4010-992d-1f0665467ea7.PNG)
And if I choose the disk directly from harvester in the node config, nothing happens and it doesn't find the path
![Uploading harv1.PNG…]()

**Comment by w13915984028** [comment]
@Vicente-Cheng Please take a look, thanks.

**Comment by Vicente-Cheng** [comment]
Hi @KarlSK8, 

Could you please describe in more detail about you mentioned above?
1. > I have personally connected a SAN to harvester then created an ext4 partition that I was able to view in the add disk view when editing.

This means you add disks that are exported by your SAN storage to the harvester node?

2. > If I go to longhorn and add a disk using its path, it says that the disk is not ready and not schedulable.

Could you try this [document](https://longhorn.io/docs/1.2.4/volumes-and-nodes/multidisk/)? To add multidisk to longhorn.

And Harvester could not use external storage as a non-system partition for virtual machines.
Reference: https://github.com/harvester/harvester/issues/2405

**Comment by KarlSK8** [comment]
Hi @Vicente-Cheng,

Thank you for reply. The document you mentioned looks helpful. I have done a similar thing when trying to add a SAN storage.
I have added the disk to the management node, then created an alias path to it (multipath). But when I went to longhorn, and added the path it's saying that it is not schedulable. What is the reason behind this in your opinion?

**Comment by Vicente-Cheng** [comment]
Hi @KarlSK8, 

Any disk added to the management node would be handled by `node-disk-manager` (called NDM).
For `mpath` device, NDM does not have the whole testing. I thought that would not be supported now.

**Comment by KarlSK8** [comment]
My question is, if I create an mpath for the SAN storage how would I mount it to be used by harvester?

**Comment by Vicente-Cheng** [comment]
Hi @KarlSK8

I think the current answer is NO.
NDM does not support mpath device now.

You could try to use w/o multipath.

**Comment by crazyelectron-io** [comment]
I have a related disk configuration/usage question.
My setup is a 3-node bare metal Harvester cluster with NVMe disks for the 'OS'. Each node has 4x 8TB SSD that I want to use as storage for Longhorn installed in a RKE2 cluster of 6 nodes running on the 3 Harvester nodes.

I am stuck at how to dedicate these SSD disks to the VM running Longhorn Pods. I don't want to use them in Harvester, just for these Longhorn nodes. As far as I can tell from the documentation I can only add volumes to VM's and not physical disks.

**Comment by Vicente-Cheng** [comment]
hi @crazyelectron-io, 

I saw you open another issue (https://github.com/harvester/harvester/issues/2507) for it.
Looks like we could discuss the above issue instead here.

**Comment by crazyelectron-io** [comment]
@Vicente-Cheng works for me... :-)
Since I also learned about the Harvester CSI driver, I figured it would be best for my setup to add the SSDs to the Harvester hosts and create Longhorn volumes to be used in the RKE2 Cluster nodes/pods through the Harvester CSI driver which is now installed by default. But since the documentation about using the driver is a bit 'dense' I could need some pointers.

Specific question I have: What are the requirements (if any) for the storage class and namespace used for the volumes created in Harvester to be 'picked' up by the Harvester CSI driver in the RKE2 cluster? Or is it just a matter of defining the PV/PVC parameters with whatever storage class or namespace used to create the volumes?
